re-organize docker cloud topics, public intro to swarm mode

londoncalling opened the pull request:
First drafts of topic updates and re-organization for Docker Cloud public intro of swarm mode beta.

Signed-off-by: Victoria Bialas <victoria.bialas@docker.com>

londoncalling commented:
@pkennedyr Hi Ryan, You can take a look at the first draft of the Docker Cloud swarm docs re-org here, and see what you think. 

The Netlify preview under **All checks have passed** ^^ takes you to the top of the docs, but you can go directly to cloud swarm docs **[here](https://deploy-preview-1669--docker-docs.netlify.com/docker-cloud/cloud-swarm/)**.

The only thing missing is adding these topics to the left side TOC menu, which I can do as soon as you think these are ready for prime time.

pkennedyr commented on docker-cloud/cloud-swarm/create-cloud-swarm.md:
```diff
@@ -4,10 +4,7 @@ keywords: swarm mode, swarms, create swarm, Cloud
-{: style="text-align:center" }
-
---------
+> **Note:** All Swarm management features in Docker Cloud are free while in Beta.
```
Let's move "free while in beta" reference to the "Using swarm mode with Docker Cloud" doc (so as to qualify that all swarm mode functionality is free while in beta).

pkennedyr commented on docker-cloud/cloud-swarm/index.md:
```diff
@@ -2,164 +2,27 @@
 ---
 
-<br>
+Docker Cloud now allows you to connect to clusters of Docker Engines running in v1.13 [swarm mode](/engine/swarm/).
```
Let's add a bit more context to the introduction of Swarm mode with something like:

_"Docker Cloud now allows you to easily manage a fleet of swarms by connecting to clusters of Docker Engines running in v1.13 [swarm mode](/engine/swarm/). 

With Beta Swarm mode in Docker Cloud, you can provision Swarms to popular Cloud providers, or register existing Swarms to Docker Cloud. Use your Docker ID to authenticate and securely access personal or team Swarms."_

Also, let's remove the below image.

pkennedyr commented on docker-cloud/cloud-swarm/index.md:
```diff
@@ -2,164 +2,27 @@
+* [Connect to a swarm through Docker Cloud](connect-to-swarm.md)
 
-![the Swarm Mode toggle](images/swarm-toggle.png)
+* [Unregister a swarm from Docker Cloud](register-swarms.md#unregister-a-swarm-from-docker-cloud)
```
Remove the following items from the navigation list:
* "Swarm statuses"
* "Unregister"
* "Reconnect"

pkennedyr commented on docker-cloud/cloud-swarm/register-swarms.md:
```diff
@@ -0,0 +1,103 @@
+title: Register existing swarms
+---
+
+At this time, you cannot _create_ swarms from within Docker Cloud. However you can register existing swarms as part of the beta functionality.
```
Change: "At this time, you cannot create swarms from within Docker Cloud. However you can register existing swarms as part of the beta functionality."

To: "With Beta Swarm mode, you can register existing swarms with Docker Cloud in order to easily manage a large number of Swarms running anywhere."

pkennedyr commented on docker-cloud/cloud-swarm/using-swarm-mode.md:
```diff
@@ -0,0 +1,58 @@
+### New to Docker Cloud?
+
+If you just signed up as a new Docker Cloud user, Swarm mode is baked into the
+standard web interface. This is leading edge functionality for Docker Cloud and
```
Remove: "This is leading edge functionality for Docker Cloud and we plan to build out swarm management features using this model in future releases."

pkennedyr commented:
cc: @borjaburgos @KickingTheTV

londoncalling commented:
@pkennedyr @borjaburgos @KickingTheTV I've incorporated Ryan's comments on the first draft of this. Just a few questions/comments for Ryan:

1.  Wondering if you would prefer a different screen snap for the swarms home page (docker-cloud/cloud-swarm/index.md) rather than removing it altogether - I thought you had wanted a screen snap there for the intro? If we got a picture of a Cloud account as it would appear to a new user (without the Swarm Mode toggle), would that be more useful? Or do you just not want an image there at all?

2. Generally, throughout the engine docs, `swarm mode` is lower case. But I went ahead and made this consistent throughout these Cloud docs as `Swarm Mode` since it seems that we are generally referring to the name of on the Docker Cloud UI, which is in fact initial caps. What do you think?

3. Tangent: On my own experiments with registering swarms in my Docker Cloud account,  I've run into some weird glitches; e.g., I can successfully register the swarm and connect to it, but soon after, it drops off in the sense that it's status shows as `UNREACHABLE` (even though the swarm is still running and reachable through the CLI). Any thoughts?

pkennedyr commented on docker-cloud/cloud-swarm/using-swarm-mode.md:
```diff
@@ -0,0 +1,60 @@
+
+## Where to go next
+
+Learn how to [register an existing swarm](register-swarms.md).
```
Technically, there are two paths users can follow next, so let's also include a reference to:

"Learn how to connect to a swarm through Docker Cloud."

pkennedyr commented on docker-cloud/cloud-swarm/create-cloud-swarm.md:
```diff
@@ -47,3 +42,10 @@ forwards your commands directly to the Docker instances running in the swarm.
+
+## Where to go next
+
+You can get an overivew of topics on [swarms in Docker Cloud](index.md).
```
Let's also include a reference to connecting to the swarm:

> Learn how to connect to a swarm through Docker Cloud.

pkennedyr commented on docker-cloud/cloud-swarm/create-cloud-swarm.md:
```diff
@@ -4,11 +4,6 @@ keywords: swarm mode, swarms, create swarm, Cloud
---------
-
 You can now create _new_ Docker Swarms from within Docker Cloud as well as
 register existing swarms.
```
Let's link "register existing swarms" to the respective doc.

pkennedyr commented:
Hey @londoncalling,

Thanks for the updates - adding replies below to your earlier comments:

1. The Swarm mode UI is pretty bland during the beta (i.e. offers a simple list view of swarms that are registered with Cloud), so let's leave out the screenshot out for now.

2. Laura originally suggested referring to "Swarm Mode" (in context of Docker Cloud) as uppercase in the docs and the UI, so I'm thinking we just continue with the uppercase reference unless you see a strong reason to change.

3. When you register a swarm without specifying an internal_endpoint, we get the IP address from the request. So if you are in the office from your laptop, that IP address will be the router of the building and therefore when trying to contact your swarm, it fails I think you have to use some software to expose your local swarm port (like ngrok) and update the internal_endpoint att of the swarm to let DC know where to ping.

londoncalling commented:
@pkennedyr that all makes sense :-) thanks for the reviews

As we discussed, next step here will be for me to add in these topics to the IA (left side menu TOC), along with further review comments and improvements.

pkennedyr commented on docker-cloud/cloud-swarm/index.md:
```diff
@@ -2,164 +2,29 @@
+* [Connect to a swarm through Docker Cloud](connect-to-swarm.md)
 
-You can switch between node cluster and swarm mode at any time, and enabling swarm mode does _not_ remove or disconnect existing node clusters.
+* [Create a new swarm in Docker Cloud](create-cloud-swarm.md)
```
Remove duplicate reference to "Create a new swarm in Docker Cloud"

sanscontext commented on docker-cloud/cloud-swarm/register-swarms.md:
```diff
@@ -0,0 +1,105 @@
+---
+
+With Beta Swarm Mode, you can register existing
+swarms with Docker Cloud in order to easily
```
s/in order to/to
s/a large number/many (or maybe "multiple") 

(this is what happens when you read FB comments on your day off. ;) )

londoncalling commented:
@pkennedyr per our discussion on Friday, I've done a first pass of the following:

1. integrated Swarm mode topics into the IA (left side topic nav menus)
2. Folded these "non-Swarm mode" topics as sub-topics under the heading "Standard Cloud Node and App Management" (we can name this anything you like, just trying this on for size)
  - Getting Started
  - Managing Applications
 - Cloud stack file YAML
3. Re-factored the Cloud welcome page (About Docker Cloud) to downplay non-Swarm topics and feature Swarm mode. 

Have a look at this when you get a chance, and let me know what you think, and we can go from there.

pkennedyr commented on _data/toc.yaml:
```diff
@@ -1027,18 +975,87 @@ manuals:
+  - sectiontitle: Standard Node and App Management
+    section:
+    - path: /docker-cloud/standard/
+      title: Standard Cloud Node and App Management
```
Let's change this title to: "Manage Infrastructure and Applications (Standard)"

pkennedyr commented on _data/toc.yaml:
```diff
@@ -1027,18 +975,87 @@ manuals:
+          title: Stackfiles for your service
+        - path: /docker-cloud/getting-started/deploy-app/12_data_management_with_volumes/
+          title: Data management with Volumes
+    - sectiontitle: Manage Applications
```
Let's move the top-level "Manage Infrastructure" into this section and have it appear directly below "Getting Started".

i.e.
>  Manage Infrastructure and Applications (Standard)
>>  Infrastructure and Application Management
>> Getting Started
>> Manage Infrastructure
>> Manage Applications
>> Cloud Stack file YAML reference

pkennedyr replied:
Alternatively, we could change the section name to "Manage Infrastructure (Standard Mode)" and move it below "Manage Swarms (Beta Swarm Mode)".

The current "Manage Infrastructure" section is highly specific to the standard mode functionality and most of the content doesn't apply to Swarm mode.

pkennedyr commented on _data/toc.yaml:
```diff
@@ -1027,18 +975,87 @@ manuals:
-    path: /apidocs/docker-cloud/
-  - title: Cloud stack file YAML reference
-    path: /docker-cloud/apps/stack-yaml-reference/
+  - sectiontitle: Standard Node and App Management
```
Let's change this title to: "Manage Infrastructure and Applications (Standard-Mode)"

pkennedyr commented on docker-cloud/standard/index.md:
```diff
@@ -0,0 +1,19 @@
+nodes, services, and applications in Docker Cloud.
+
+* [Getting started with Docker Cloud](/docker-cloud/getting-started/index.md)
+
```
Add "Infrastructure Management" section below "Getting Started".

pkennedyr commented on docker-cloud/standard/index.md:
```diff
@@ -0,0 +1,19 @@
+title: Standard Methods for Managing Nodes and Apps in Docker Cloud
+notoc: true
+---
+
```
Remove the reference to swarm-mode beta in this section:

> Beta Feature. Swarm Mode for Docker Cloud is currently a public Beta. Swarm management features are free during the Beta.

pkennedyr commented on _data/toc.yaml:
```diff
@@ -917,72 +917,20 @@ manuals:
+      title: Connect to a swarm through Docker Cloud
+    - path: /docker-cloud/cloud-swarm/link-aws-swarm/
+      title: Link to Amazon Web Services to create Swarms
   - sectiontitle: Manage Builds and Images
```
Elevate the "Manage Build and Images" section to appear above the "Swarm Mode (Beta)" section.

pkennedyr commented on docker-cloud/index.md:
```diff
@@ -48,26 +48,20 @@ Log in to Docker Cloud using your free [Docker ID](../docker-id/).
-  </tr>
-  <tr>
-    <td class="bluebar" width="50%"><a href="apps/index.md">Manage Applications</a></td>
+    <td class="bluebar" width="50%"><a href="cloud-swarm/index.md">Swarms in Docker Cloud (Beta)</a></td>
```
Change to:

> Manage Swarms (Beta Swarm Mode)

pkennedyr commented on docker-cloud/index.md:
```diff
@@ -48,26 +48,20 @@ Log in to Docker Cloud using your free [Docker ID](../docker-id/).
   </tr>
   <tr>
-    <td class="bluebar" colspan="2"><a href="infrastructure/index.md">Manage Infrastructure</a></td>
+    <td class="bluebar" width="50%"><a href="infrastructure/index.md">Manage Infrastructure</a></td>
```
Change to:

> Manage Infrastructure (Standard Mode)

pkennedyr commented on docker-cloud/index.md:
```diff
@@ -48,26 +48,20 @@ Log in to Docker Cloud using your free [Docker ID](../docker-id/).
   <tr>
-    <td class="bluebar" colspan="2"><a href="infrastructure/index.md">Manage Infrastructure</a></td>
+    <td class="bluebar" width="50%"><a href="infrastructure/index.md">Manage Infrastructure</a></td>
+    <td class="bluebar" colspan="2"><a href="standard/index.md">Standard Node and App Management</a></td>
```
Change to:

> Manage Nodes and Apps (Standard Mode)

pkennedyr commented on docker-cloud/index.md:
```diff
@@ -48,26 +48,20 @@ Log in to Docker Cloud using your free [Docker ID](../docker-id/).
+    <td class="plain" width="50%"><p>Link to your hosts, upgrade the Docker Cloud agent, and manage container distribution. See the <a href="infrastructure/cloud-on-aws-faq.md">AWS FAQ</a> and <a href="infrastructure/cloud-on-packet.net-faq.md">Packet.net FAQ</a></p></td>
+    <td class="plain" width="50%"><p>Deploy and manage nodes, services, and applications in the original Docker Cloud model (pre-Swarm Mode). Includes <a href="getting-started/index.md">getting started tutorial</a>, <a href="apps/index.md">managing applications</a>, and the <a href="/docker-cloud/apps/stack-yaml-reference.md">Cloud stack file reference</a>, all applicable only to non-Swarm Mode t ...
   </tr>
   <tr>
```
In the "About Docker Cloud" section appearing below the diagram, can we also:

A) Include an equivalent section on Swarms
B) Qualify the difference between "Swarm Mode" and "Standard Mode" in applicable sections, e.g.:
- Manage Swarms (Swarm Mode)
- Infrastructure Management (Standard Mode)
- Services, Stacks, and Applications (Standard Mode)


These topics cover the traditional, pre-Swarm model for deploying and managing nodes, services, and applications in Docker Cloud.

pkennedyr replied:
Also, let's change the order of the diagram to:

- Manage Builds and Images
- Manage Infrastructure
- Manage Swarms (Beta Swarm Mode)
- Manage Nodes and Apps (Standard Mode)

pkennedyr commented:
Thanks, @londoncalling - these are nice improvements!

I've added a few comments to make the delineation b/w "Swarm Mode" and "Standard Mode" a bit more clear. Feel free to have a look and let me know if you have any comments/questions.

pkennedyr commented on docker-cloud/cloud-swarm/using-swarm-mode.md:
```diff
@@ -0,0 +1,61 @@
+The current release of Docker Cloud automatically opts in new users to Beta
+Swarm Mode.
+
+> **Note:** All Swarm management features in Docker Cloud are free while in Beta.
```
Remove additional reference to swarm mode being free while in beta:

>  **Note:** All Swarm management features in Docker Cloud are free while in Beta.

londoncalling commented:
@pkennedyr @mstanleyjones @johndmulhausen (docs cc'ed as a heads-up for  docs infrastructure point `#2` below)

Hi Ryan, Thanks, great comments, I think it's getting there.
 
In this round:

1. incorporated your review comments 
- Cloud Home page diagram better clarifies Swarm mode vs. Standard mode (using consistent wording and structure)
- Same with left nav menus
- Removed "free" statement from the Swarm mode Beta banner at the top of those pages

2. Added a new type of "advisory" called `previewdocs` which uses `previewflag` in the topics that need it. This enables us to use a green bar message at the top of our Swarm Mode `Beta` docs instead of the yellow bar. (We use the yellow bar banners in other docs like Toolbox, Kitematic, and old stand-alone Swarm to indicate legacy docs, so I didn't want to cue people on a Beta with that subliminal idea. I figured green for go, yellow for slow down/old stuff.)

3. A good place to start this review on the Netlify preview:

- [Cloud Home topic with diagram](https://deploy-preview-1669--docker-docs.netlify.com/docker-cloud/)

- [Swarms in Docker Cloud overview page](https://deploy-preview-1669--docker-docs.netlify.com/docker-cloud/cloud-swarm/)

pkennedyr commented:
Thanks, @londoncalling. Applied a few other comments:

- Let's change the "Manage Infrastructure" section name to "Manage Infrastructure (Standard Mode)" and move it below "Manage Swarms (Beta Swarm Mode)". The current "Manage Infrastructure" section is highly specific to the standard mode functionality and most of the content doesn't apply to Swarm mode.
- Images appear to be missing on the following pages:
-- Builds and images overview (docker-cloud/builds/)
-- Docker Cloud repositories (docker-cloud/builds/repos/)
- re: /orgs page, I don't believe a `Read` permission option is available for Swarms (only `Admin`) ... @fermayo can you confirm? If so, please remove `Read` and change the `Admin` definition to: "View swarms, manage swarms, add users"

londoncalling commented:
@pkennedyr 

Thanks, I pushed more changes just now:

1.  "Manage Infrastructure" --> "Manage Infrastructure (Standard Mode)" and moved to below "Manage Swarms (Beta Swarm Mode)" in the left nav menu. Also renamed on the Welcome page diagram, but I haven't moved it in the table yet. Not sure about how to restructure that table any more or if it's needed.

2. I do see the missing images on Netlify preview but I haven't touched the images or the calls to them, and they render on local builds. Let's see if this doesn't iron itself out, otherwise I'll have another look tomorrow.

3. I went ahead and changed the description of the permissions on Swarms per your comments, and can modify more if needed.

That's all for now, we can re-group tomorrow on whatever else needs to be done.

londoncalling commented:
Okay, I'm stumped by the image display problem in `docker-cloud/builds` and `docker-cloud/builds/repos` (same image not displaying in both files). It is still not showing up on the Netlify preview, but it is there in file in the branch (https://github.com/londoncalling/londoncalling.github.io/blob/cloud-docs-with-swarm/docker-cloud/builds/index.md). I'll have a look tomorrow.

londoncalling commented:
Re: the missing image , I'll fix it later today by just creating a new one with a different name .

pkennedyr commented:
Thanks, @londoncalling. The latest changes look great. 

Re: the missing images, I'm seeing random examples (e.g. on the Welcome screen), so not sure what is causing this either. I think we're good to go, once these are addressed

Thanks again and appreciate all your help!

pkennedyr commented on docker-cloud/index.md:
```diff
@@ -48,26 +48,20 @@ Log in to Docker Cloud using your free [Docker ID](../docker-id/).
   </tr>
   <tr>
-    <td class="plain" colspan="2"><p>Learn how to link to your hosts, upgrade the Docker Cloud agent, and manage container distribution. See the <a href="infrastructure/cloud-on-aws-faq.md">AWS FAQ</a> and <a href="infrastructure/cloud-on-packet.net-faq.md">Packet.net FAQ</a></p></td>
+    <td class="plain" width="50%"><p>Link to your hosts, upgrade the Docker Cloud agent, and manage container distribution. See the <a href="infrastructure/cloud-on-aws-faq.md">AWS FAQ</a> and <a href="infrastructure/cloud-on-packet.net-faq.md">Packet.net FAQ</a></p></td>
```
Change the "Manage Swarms (Beta Swarm Mode)" description to:

> Provision swarms to popular cloud providers, register existing swarms, and use your Docker ID to authenticate and securely access personal or team swarms.

pkennedyr commented on docker-cloud/index.md:
```diff
@@ -48,26 +48,20 @@ Log in to Docker Cloud using your free [Docker ID](../docker-id/).
   <tr>
-    <td class="plain" colspan="2"><p>Learn how to link to your hosts, upgrade the Docker Cloud agent, and manage container distribution. See the <a href="infrastructure/cloud-on-aws-faq.md">AWS FAQ</a> and <a href="infrastructure/cloud-on-packet.net-faq.md">Packet.net FAQ</a></p></td>
+    <td class="plain" width="50%"><p>Link to your hosts, upgrade the Docker Cloud agent, and manage container distribution. See the <a href="infrastructure/cloud-on-aws-faq.md">AWS FAQ</a> and <a href="infrastructure/cloud-on-packet.net-faq.md">Packet.net FAQ</a></p></td>
+    <td class="plain" width="50%"><p>Deploy and manage nodes, services, and applications in the original Docker Cloud model (pre-Swarm Mode). Includes <a href="getting-started/index.md">getting started tutorial</a>, <a href="apps/index.md">managing applications</a>, and the <a href="/docker-cloud/apps/stack-yaml-reference.md">Cloud stack file reference</a>, all applicable only to non-Swarm Mode t ...
```
Change the long description under "Manage Nodes and Apps (Standard Mode)" to:

> Deploy and manage nodes, services, and applications in Docker Cloud (Standard Mode).

pkennedyr commented:
Applied a couple content updates to docker-cloud/ diagram (i.e. swarm mode referenced the same content as the infrastructure section).

londoncalling commented:
Hi @pkennedyr ,
Okay, I think i've incorporated all your latest comments:

1. Modified descriptions of Swarm Mode and Standard Mode on the [Cloud Welcome page](https://deploy-preview-1669--docker-docs.netlify.com/docker-cloud/). (Yes I had mistakenly copied content from one place in that table to another and forgot to re-do it 😕  .. Good catch 🙂 )

2. Fixed repository image to render properly in "build" topics:  [Builds and images overview](https://deploy-preview-1669--docker-docs.netlify.com/docker-cloud/builds/) and [Docker Cloud repositories](https://deploy-preview-1669--docker-docs.netlify.com/docker-cloud/builds/repos/)
(The image itself on GitHub repo must have been corrupted. I took a new screen snap and now it renders properly with the same image name.)

I _think_ we are in good shape for tomorrow, and good to go. Just let me know if you and your team have any other comments, and when you want me to publish these docs.

Vicky

londoncalling commented:
LGTM, merging to publish

@pkennedyr
